cloudferry/actions/block_storage/migrate_volumes.py

```diff
@@ -215,13 +249,46 @@ def run(self, **kwargs):
         view = VolumeMigrationView(volumes)
         view.initial_message()
 
-        for i, volume in enumerate(volumes):
-            view.before_migration(i, volume)
-
-            migrated_volume = self.migrate_volume(volume)
-
-            view.after_migration(i, volume)
-            new_volumes[volume['id']] = migrated_volume
+        for i, src_volume in enumerate(volumes):
+            view.before_migration(i, src_volume)
+
+            LOG.info("Checking if volume '%s' already present in destination",
+                     volume_name(src_volume))
+            dst_cinder = self.dst_cloud.resources[utils.STORAGE_RESOURCE]
+
+            dst_volume = dst_cinder.get_migrated_volume(src_volume['id'])
```

Comment:
You'd need to modify `get_migrated_volume` to look for volume with the same ID for 'shared-nfs' case, and keep existing implementation for other backends.